File: Arrays/remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.go
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
package main

func removeDuplicates(nums []int) int {
    if len(nums) == 0 {
        return 0
    }

    distinctCount := 1
    for i := 1; i < len(nums); i++ {
        if nums[i] != nums[distinctCount-1] {
            nums[distinctCount] = nums[i]
            distinctCount++
        }
    }

    return distinctCount
}

File: Arrays/remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.rs
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/

pub struct Solution {}
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        if nums.is_empty() {
            return 0;
        }

        let mut distinct_count: usize = 1;
        
        for i in 1..nums.len() {
            if nums[i] != nums[distinct_count - 1] {
                nums[distinct_count] = nums[i];
        distinct_count += 1;
      }
    }

    distinct_count as i32
  }
}

fn main() {
    let mut nums = vec![1, 1, 2];
    let result = Solution::remove_duplicates(&mut nums);
    println!("{}", result);
}
